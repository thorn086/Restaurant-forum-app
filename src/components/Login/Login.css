.login-info-loc{
  border: 1px solid black;
width: 90%;   
background-color: rgba(70, 70, 193, 0.9);
margin:1rem;
border-radius: 5%;
box-shadow: 3px 3px rgb(61, 61, 95);
padding: 1rem;
text-align: center;
height: auto;
letter-spacing: 2px;
color: white;
font-size: 16px;
font-weight: 600;
text-shadow: black 1px 1px;
line-height: 2;
}
.error-message{
height: auto;
background-color: rgb(1, 1, 245);
width: auto;
color:yellow;
border-radius: 1rem;
text-shadow: black -1px -1px;
text-transform: uppercase;
font-size: 20px;
text-align: center;
padding: 1rem;
margin: 1rem;
}
#username-login, #password-login{
  width: 15rem;
  height: 2rem;
}

@media (min-width: 420px){
  .login-info-loc{
    font-size: 17px;
  }
#username-login, #password-login{
  width: 20rem;
  height: 2rem;
  margin:1rem .5rem;
}
.back-btn{
  margin-left: 6rem;
}
}
@media (min-width: 544px){
  .login-info-loc{
    font-size: 17px;
  }
#username-login, #password-login{
  width: 22rem;
  height: 2rem;
  margin:1rem .5rem;
}
.back-btn{
  display: inline-block;
  margin: 1rem 0 1rem 2rem;
}
}
@media (min-width: 760px){
  .login-info-loc{
    font-size: 17px;
    text-align: left;
    margin-left: 2rem;
    margin-top: 2rem;
    padding-left: 3rem;
  }
#username-login, #password-login{
  width: 22rem;
  height: 2rem;
  margin:1rem .5rem;
}
#password-login{
  margin-left: 3.5rem;
}
.back-btn{
  display: inline-block;
  margin: 1rem 0 1rem 6rem;
}
}
@media (min-width: 1024px){
  .state_list{
    width: 90%;
    margin: auto;
  }
  #log-in{
    width: 80%;
    margin:auto;
  }
  #password-login{
    margin-left: 4.5rem;
  }
  #username-login{
    margin-left: 1.7rem;

  }
}
@media (min-width: 1440px){
  

  .login-info-loc{
    font-size: 20px;
  }
  #password-login{
    margin-left: 4.5rem;
  }
  #username-login{
    margin-left: 1.2rem;

  }
  .back-btn{
    display: inline-block;
    margin: 1rem 0 1rem 8rem;
  }
}
@media (min-width: 1600px){
 .state_list{
   width: 60%;
   margin: auto;
 }
 .login-info-loc{
  font-size: 22px;
  width: 100%;
}
.back-btn{
  display: inline-block;
  margin: 1rem 0 1rem 6rem;
}
}