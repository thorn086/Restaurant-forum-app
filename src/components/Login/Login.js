import React from 'react';
import './Login.css';
import restForumContext from '../../context';
import NavBar from '../../components/NavBar/NavBar';
import TokenService from '../../services/token-services';
import ApiAuthService from '../../services/api-auth-service';
class Login extends React.Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  }
  state = {error: null }
  static contextType = restForumContext

  handleLoginSuccess = (email, userId) => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/states'
    history.push(destination)
    this.context.userInfo({ email, userId })

  };
  //checks user has authorization to enter site
  handleSubmitWithAuth = e => {
    e.preventDefault()
    this.setState({ error: null });
    const { email, password } = e.target
    ApiAuthService.postLogin({
      user_email: email.value,
      password: password.value
    })
      .then(res => {
        TokenService.saveAuthToken(res.authToken)
        TokenService.saveUserId(res.userId)
        this.handleLoginSuccess(email.value, res.userId)
        email.value = ''
        password.value = ''
      })
      .catch(res => {
        this.setState({ error: res.error });
      })
  };

  //validates information is correct and displays to DOM
  handleInccorectLogin() {
    if (this.state.error === null) {
      return (
        <div></div>
      )
    } else {
      return (
        <div className='error-message'>
          <strong>
            {this.state.error}
          </strong>
        </div>
      )
    }
  }
  render() {

    return (
      <div className='login-form'>
        <h2 className='state_title'>Login Form</h2>
        <NavBar />
        <div className='state_list'>
          <form
            id='log-in'
            onSubmit={this.handleSubmitWithAuth}
          >
            <div className="login-info-loc">
              <div className='username-login'>
                <div className='login_demo-creds'>
                  <strong>
                    Demo credentials: restapp@gmail.com / Food@ful1#
                  </strong>
                </div>
                <label htmlFor='username-login'>Email Address</label>
                <input type='email' name='email' id='username-login' required />
              </div>
              <div className='password-login'>
                <label htmlFor='password-login'>Password</label>
                <input type='password' name='password' id='password-login' required />
              </div>
              <div>
              </div>
              {this.handleInccorectLogin()}
            </div>
            <button className='back-btn' type='submit' value='LogIn'>
              Log In
                </button>
            <button className='back-btn' tag='button' onClick={() => this.props.history.goBack()}>Back</button>



          </form>
        </div>
      </div>

    )
  };

}
export default Login