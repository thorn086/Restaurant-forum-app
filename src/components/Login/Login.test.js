import React from 'react';
import Login from './Login';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Login', () => {
    const login = shallow(<Login/>);
  
    it('renders properly', () => {
        expect(login).toMatchSnapshot();
    });
});