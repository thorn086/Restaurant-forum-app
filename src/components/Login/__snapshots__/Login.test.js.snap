// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders properly 1`] = `ShallowWrapper {}`;
